sensor/common/clusterentities/store_test.go

```diff
@@ -97,3 +102,80 @@ func (s *ClusterEntitiesStoreTestSuite) TestMemoryWhenGoingOffline() {
 		})
 	}
 }
+
+func TestEntityData_GetDetails(t *testing.T) {
+	tests := map[string]struct {
+		edFun            func() *EntityData
+		wantContainerIDs []string
+		wantPodIPsSorted []net.IPAddress
+	}{
+		"Single values": {
+			edFun: func() *EntityData {
+				ed := &EntityData{}
+				ed.AddIP(net.ParseIP("10.0.0.1"))
+				ed.AddContainerID("abc", ContainerMetadata{})
+				return ed
+			},
+			wantContainerIDs: []string{"abc"},
+			wantPodIPsSorted: []net.IPAddress{net.ParseIP("10.0.0.1")},
+		},
+		"Multiple sorted values": {
+			edFun: func() *EntityData {
+				ed := &EntityData{}
+				ed.AddIP(net.ParseIP("10.0.0.1"))
+				ed.AddIP(net.ParseIP("10.0.0.2"))
+				ed.AddContainerID("abc", ContainerMetadata{})
+				ed.AddContainerID("def", ContainerMetadata{})
+				return ed
+			},
+			wantContainerIDs: []string{"abc", "def"},
+			wantPodIPsSorted: []net.IPAddress{net.ParseIP("10.0.0.1"), net.ParseIP("10.0.0.2")},
+		},
+		"Multiple unsorted values": {
+			edFun: func() *EntityData {
+				ed := &EntityData{}
+				ed.AddIP(net.ParseIP("10.0.0.9"))
+				ed.AddIP(net.ParseIP("10.0.0.2"))
+				ed.AddContainerID("abc", ContainerMetadata{})
+				ed.AddContainerID("def", ContainerMetadata{})
+				return ed
+			},
+			wantContainerIDs: []string{"abc", "def"},
+			wantPodIPsSorted: []net.IPAddress{net.ParseIP("10.0.0.2"), net.ParseIP("10.0.0.9")},
+		},
+		"Invalid IP": {
+			edFun: func() *EntityData {
+				ed := &EntityData{}
+				ed.AddIP(net.ParseIP("foo.bar.baz.boom"))
+				ed.AddIP(net.ParseIP("10.0.0.2"))
+				ed.AddContainerID("abc", ContainerMetadata{})
+				return ed
+			},
+			wantContainerIDs: []string{"abc"},
+			wantPodIPsSorted: []net.IPAddress{net.ParseIP("10.0.0.2")},
+		},
+		"No Container ID": {
+			edFun: func() *EntityData {
+				ed := &EntityData{}
+				ed.AddIP(net.ParseIP("10.0.0.1"))
+				return ed
+			},
+			wantContainerIDs: []string{},
+			wantPodIPsSorted: []net.IPAddress{net.ParseIP("10.0.0.1")},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ed := tt.edFun()
+			gotContainerIDs, gotPodIPs := ed.GetDetails()
+			// Sort as GetDetails is not guaranteed to return sorted data.
```

Comment:
If not guaranteed, why does `GetDetails` try to sort containers? And not IPs?

Reply:
I answered the same question to Guzman already:

The reason why I am sorting here is that `maps.Keys` returns an iterator and we want to have a slice. If you read https://github.com/golang/go/issues/61900, then Ross said that 
> In particular, we expect that `x := slices.Sorted(maps.Keys(m))` will be a common pattern.

I applied it here.

Reply:
Now the consequence is that the test-case `"Multiple unsorted values":` will be flaking. Do you want me to sort the value returned from `GetDetails` in the test or to remove that test case?